l10n_it_intrastat_statement/tests/test_intrastat_statement.py

```diff
@@ -163,6 +163,30 @@ def test_statement_sale_quarter(self):
         # Last line is section line, for quarter report it should be 64 chars
         self.assertEqual(len(file_content.splitlines()[-1]), 64)
 
+    def test_statement_sale_quarter_adjustment_period(self):
+        """Check that sale adjustments can be in a different period type"""
+        invoice = self._get_intrastat_computed_invoice()
+        month = fields.Date.from_string(invoice.date_invoice).month
+        quarter = 1 + (month - 1) // 3
+        statement = self.statement_model.create({
+            'period_number': quarter,
+            'period_type': 'T',
+        })
+        statement.compute_statement()
+
+        adjustment_vat_code = 'adjust_vat'
+        statement.sale_section2_ids = [(0, 0, {
+            'partner_id': invoice.partner_id.id,
+            'vat_code': adjustment_vat_code,
+            'sign_variation': '+',
+            'year_id': fields.Date.from_string(invoice.date_invoice).year,
+            'month': 1,
```

Comment:
`month` perhaps?

Reply:
Here I am testing that the adjustment can be in a different period type than the `statement` so it doesn't have to be related to the `statement`'s month.
`month` is only used to compute the `quarter` of the `statement`.

Reply:
In this way (putting always 1 as `month`), in the first three month of the year month would be in the same quarter of statement, is it wanted?

Reply:
The code now allows the period stated in the adjustment to be independent from the period of the `statement` and this is what this test case is checking.
Anyway, I understand that this wouldn't be a realistic use-case when they are overlapping, so I changed the adjustment's year; can you check?

BTW, thanks for the review :)

Reply:
Ok (I don't use this version anyway)